### ISSUE TITLE
[DOC] API reference should point more clearly to code/import location

### ISSUE THREAD

**Issue by fkiraly** [alan-turing-institute/sktime#2413]
Currently, the API reference does not clearly point to the code location of the class/method documented.

Some classes will have examples with an import statement, but in general this is not the case.

Hence, a typical user who does not have a developer install will be unable to find the correct import command from looking at the API reference alone.

This should be addressed, e.g., by a link provided to source and/or a printed full import path at the top of each method/class page in the API reference.

**Comment by aiwalter** [comment]
yes good point, that is why I recently added docstring examples to the datasets. Ideally there can be a section `Import` in the docstring where the path is given? Not sure if this can work with Sphinx

**Comment by fkiraly** [comment]
this is channeling user frustration from users at my workplace 😄

**Comment by lmmentel** [comment]
Can you give an example of what you mean or point to a specific place in the docs and describe the expected outcome?

**Comment by fkiraly** [comment]
ok, let's say I'm browsing the docs and find `AutoARIMA`
https://www.sktime.org/en/latest/api_reference/auto_generated/sktime.forecasting.arima.AutoARIMA.html

and say, "nice, this would be a great forecaster to use!"

Then, the next step would be to write sth like `from ???? import AutoARIMA`, with the missing information being the `????`.
It is unclear from a user perspective how to get from the doc page to the ????, and to my understanding there is no general way.

This is how the user experience looks like:
![image](https://user-images.githubusercontent.com/7985502/162992091-f24b610f-09e4-4ced-a135-38c09d001427.png)

The "natural" places where a user might look, per expectation:

* on the left, in the navigation bar. That is not ordered by module name level, so doesn't help.
* at the top, e.g,. for a header `forecasting.arima.AutoARIMA`

The shortest path to the `????` information is clicking on the `[source]`, then scrolling up until the file path gets visible in the GitHub interface which is not at all obvious imo.

**Comment by aiwalter** [comment]
`sklearn` and `pandas` have it in header, so might be worth to check how they did it

**Comment by aiwalter** [comment]
what we want it this:
![image](https://user-images.githubusercontent.com/29627036/163812361-ce26adc0-2aa7-4526-b8ba-8cb0081c9ff4.png)
 I made a draft PR but this headline did not yet show up 😞
